### ISSUE TITLE
[GamePhysics] Displacement Unit Vector DD?

### ISSUE THREAD

**Issue by samm82** [JacquesCarette/Drasil#1814]
Should there be a DD added for the displacement unit vector, using the information below? This seems like it is similar from the DDs in Projectile.

![dd?](https://user-images.githubusercontent.com/35857611/62738330-15cc7d00-ba00-11e9-94f7-212bba792473.png)

**Comment by oluowoj** [comment]
@samm82, I guess yes. Do you have this as a DD in the Projectile SRS?

**Comment by samm82** [comment]
This exact DD isn't in Projectile, but there are similar ones:
![velMag](https://user-images.githubusercontent.com/35857611/62886572-5cc0b800-bd09-11e9-98e0-bb2afcf8d517.png)

**Comment by oluowoj** [comment]
ha I see, I get.

**Comment by smiths** [comment]
@samm82, yes, DDs for the magnitude of a vector and for the definition of a unit vector would be good additions to the Drasil knowledge base.

**Comment by samm82** [comment]
When you say "additions to the Drasil knowledge base", do you mean that I should make general DDs that should be in `drasil-data`? If so, I agree, and I believe we discussed this in a previous issue. Where in the data package do you think they should go? There is currently a `Theories` directory that contains the `Physics` module - a DD technically isn't a theory, so I don't think this folder makes the most sense for it. Any ideas? @smiths

**Comment by smiths** [comment]
@samm82, yes, more general DDs would be good.  They don't belong in `Physics` because these ideas (magnitude and unit vector) are mathematical ideas.  Is there a Math module?  I haven't kept up on the organization of the data packages.  What is your recommendation?  We can get @JacquesCarette to say whether your proposal fits with the current hierarchy.

**Comment by bmaclach** [comment]
@samm82 I was the one who named that "Theories" directory and I intended it as a place for any general TMs, GDs, DDs, or IMs. I thought "Theories" would be a general enough name to encompass them all, though it might be better to split each type of "theory" into a different file, given the confusion. Or potentially rename it to something less misleading.

**Comment by bmaclach** [comment]
(Though with a proper "theory" type in the future from which TMs, GDs, and DDs can all be built, I do think it makes sense to store them all in a "Theories" directory, which is why I named it that).

**Comment by samm82** [comment]
Fair enough - I just thought about that before reading your comment 😄 I'll add a `Math` module there

**Comment by smiths** [comment]
@bmaclach, I like the name Theories (unless we think of something better).  :-)  Something that is a DD in one example might be a TM in another.  Whether it is a TM or a DD depends on the specific perspective in a given example.  They really are all theories (in a non-mathematical use of the word).  Maybe we'll start using the word model more in the future?

**Comment by samm82** [comment]
Should the vector magnitude DD from Projectile be specific to speed = |velocity|, or should it be generalized to scalar = |vector|, and then we can use it like "*v* comes from applying DD: vecMag to *__v__*"?

**Comment by samm82** [comment]
Same question for the creation of the new DD - should it use displacement, or a generic placeholder vector? @smiths

**Comment by smiths** [comment]
Definitely generalized.  The concept of magnitude and unit vector are the important generic parts.  They already come up in multiple examples (for different vectors).

**Comment by samm82** [comment]
Should we have a generic unit as well, to show that the quantities aren't unitless without constraining them to be a certain type?

I'm thinking of a unit with the name "generic unit" that measures "quantity" and has short form "unit". Thoughts? @smiths

**Comment by smiths** [comment]
I can see what you are getting at @samm82, but I'm not sure about a generic unit.  I think we would need to put some design into this (a type system for units?).  I don't think this is something we should throw together quickly.  Hopefully you can do the definition without mentioning units.

**Comment by samm82** [comment]
Currently, as shown in the following DD, units are always displayed (unless there is the decision made to not display units throughout the entire document). 

![velMag](https://user-images.githubusercontent.com/35857611/62886572-5cc0b800-bd09-11e9-98e0-bb2afcf8d517.png)

A similar issue arises for generic quantities (like a generic vector with an arbitrary symbol). How should we define these, and where? Locally in the `Theories.Math` module, or in the `Quantities.Math` module?

**Comment by smiths** [comment]
These are good questions.  In many instances we will have something abstract that isn't associated with a specific unit, but as it is refined, units will be added.  There are other cases where the generic model has units, but the specific instance has different units.  For instance, volume in 3D will change from m^3 to m^2 in 2D.  I do not have a quick answer for this.  Let us plan on discussing this during our "all hands" meeting tomorrow (Thursday).

**Comment by JacquesCarette** [comment]
`Theories` is good. There can definitely be a `Math` module under that. This is definitely moving in the right direction.

**Comment by samm82** [comment]
Also something to note about this specific issue is that this would involve a TM (Newton's Law) referencing a DD (unit vector), which I don't think is allowed currently.

**Comment by smiths** [comment]
Yes, I believe your idea is correct @samm82, although I'm not sure about how you said it.  :-)  This issue is about the displacement unit vector DD.  If we follow through on what we are discussing, then the unit vector definition cannot be a DD.  What we are discussing is applying the generic definition of magnitude, or a unit vector, and then creating a local definition for a specific problem.  This is a refinement.  We can refined theoretical models and general definitions, but as things are currently written, we cannot refine data definitions. 

I think this is what you are getting at above, although you mention Newton's Law.  I think you are thinking of another issue #1822.  We made Newton's Law a TM so that we can refine it to more specific GDs.  TMs can reference DDs, they just can't be refined into a DD.
